framework/auth/views.py

```diff
@@ -637,3 +639,80 @@ def resend_confirmation_post(auth):
 
     # Don't go anywhere
     return {'form': form}
+
+
+def oauth_user_email_get():
+    """
+    Landing view for first-time oauth-login user to enter their email address.
+    HTTP Method: GET
+    """
+
+    form = ResendConfirmationForm(request.form)
+
+    return {
+        'form': form,
+    }
+
+
+def oauth_user_email_post():
+    """
+    View to handle email submission for first-time oauth-login user.
+    HTTP Method: POST
+    """
+
+    form = ResendConfirmationForm(request.form)
+
+    session = get_session()
+    try:
+        oauth_provider = session.data['oauth_user_provider']
+        oauth_id = session.data['oauth_user_id']
+        oauth_fullname = session.data['oauth_user_fullname']
+    except KeyError:
+        raise HTTPError(http.UNAUTHORIZED)
+
+    if form.validate():
+        clean_email = form.email.data
+        user = get_user(email=clean_email)
+        oauth = {
+            oauth_provider: {
+                'oauth_id': oauth_id,
+            },
+        }
+        if user:
+            # TODO: link user's OSF account with ORCID
+            # 1. notify the user that OSF account will be linked with ORCID profile
+            # 2. user can login through both ORCID and OSF
+            # 3. send confirmation email (no need to reply)
```

Comment:
As discussed, we do need to verify here, to be sure that the person authorizing with ORCID actually has access to this email address